extensions/Worker.Extensions.EventHubs/src/EventDataConverter.cs

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core.Amqp;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Azure.Messaging.EventHubs;
+using Microsoft.Azure.Functions.Worker.Extensions.EventHubs;
+
+namespace Microsoft.Azure.Functions.Worker
+{
+    [SupportsDeferredBinding]
+    [SupportedConverterType(typeof(EventData))]
+    [SupportedConverterType(typeof(EventData[]))]
+    internal class EventDataConverter : IInputConverter
+    {
+        public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+        {
+            ConversionResult result = context?.Source switch
+            {
+                ModelBindingData binding => ConversionResult.Success(ConvertToEventData(binding)),
+                // Only array collections are currently supported, which matches the behavior of the in-proc extension.
+                CollectionModelBindingData collection => ConversionResult.Success(collection.ModelBindingDataArray
+                    .Select(ConvertToEventData).ToArray()),
+                _ => ConversionResult.Unhandled()
+            };
+            return new ValueTask<ConversionResult>(result);
+        }
+
+        private EventData ConvertToEventData(ModelBindingData binding)
+        {
+            if (binding.ContentType != Constants.BinaryContentType)
+            {
+                throw new InvalidOperationException(
```

Comment:
I think I forgot to mention this in the service bus PR as well, but we would like all the converters that use ParameterBindingData to confirm the source of the data to determine if this is the converter that should be handling the request:

```csharp
// e.g. this but with AzureEventHubsEventData
if (bindingData?.Source is not Constants.CosmosExtensionName)
{
    _logger.LogTrace("Source '{source}' is not supported by {converter}", bindingData?.Source, nameof(CosmosDBConverter));
    return ConversionResult.Unhandled();
}
```

Reply:
Does the conversion result need to be unhandled or can we throw like we do for an invalid content type?

Reply:
In this case it should be unhandled; we throw with content type because at that point we know we can handle the conversion but were not provided the correct content-type. Now that I'm writing all that out, I can see how content type could also be unhandled ☠️

Reply:
I changed it to use ConversionResult.Failed for both cases.

Reply:
I'm fine with that for now, I can discuss with the team if we want to change things across all converters for consistency and how want want to deal with all those cases

Reply:
What is the difference in behavior between Unhandled and Failure? Does Unhandled mean we will fall back to other converter (potentially defined in the host extension)? If so, I don't think we would want to do that. 
Or does it just influence the exception reported back to the user?

Reply:
I believe that unhandled will lead to falling back to other converters, however we now we have the `[AllowConverterFallback(true)]` attribute which also prompts us to fall back to converters; so I don't actually know what the behaviour looks like if we return unhandled today and AllowConverterFallback = false - will this just lead to a fail?

@surgupta-msft do you have any insights? Wondering if you tested a case like this when building the bypass deferred binding feature?

Reply:
If conversion is unhandled and AllowConverterFallback = false, then the behavior will be exactly same as it was before my changes of bypass deferred binding (if none of the registered converters can handle conversion). So, below line will be returned.

https://github.com/Azure/azure-functions-dotnet-worker/blob/54b185f0c7ed74c0db7b569021defb3dfbf05bab/src/DotNetWorker.Core/Context/Features/DefaultInputConversionFeature.cs#L86


Below is the flow of conversion logic if it helps

```csharp

// Convert using advertised converters. Return if Succeeded/Failed
// if (AllowConverterFallback == true)
       // Convert using registered converters. Return if Succeeded/Failed
// return unhandled
```